### PR TITLE
[WIP] PostActions that trigger downstream builds

### PR DESCRIPTION
**Pull request by PI-Victor**
The trello card is ON HOLD, please do not merge this PR.
This is an early pr related to https://trello.com/c/T5qfdyg9/628-8-trigger-downstream-builds-after-build-completion please do not merge.

[Review on Reviewable](https://reviewable.io/reviews/openshift/openshift-docs/1346)

### CODE REVIEW

File: dev_guide/builds.adoc [hunk 1]
```diff
@@ -87,14 +87,35 @@ image tag or the source code changes:
...
+        }
+      ],
+      "onFailure": [
+        {
+          "type": "StartBuild",
+          "build": {
+            "kind": "BuildConfig",
+            "name": "downstream-buildconfig-3",
+            "namespace": "downstream-bc-namespace"
+            },
```
Comment (php-coder) [inline dev_guide/builds.adoc]: May this extra comma lead to any problems during parsing?

File: dev_guide/builds.adoc [hunk 2]
```diff
@@ -1359,7 +1382,53 @@ Configuration change triggers currently only work when creating a new
...
+      "kind": "BuildConfig",
+       "name": "downstream-buildconfig-1", <1>
+       "namespace": "downstream-bc-namespace" <2>
+    }
+  }
+]
+----
+<1> Specify the buildconfig that should be triggered when the current image is
+successful.
+<2> Here we specify the namespace where this build should be triggered.
```
Comment (php-coder) [inline dev_guide/builds.adoc]: As far I can understand, namespace is optional, right? May we should write it explicitly?

File: dev_guide/builds.adoc [hunk 3]
```diff
@@ -1359,7 +1382,53 @@ Configuration change triggers currently only work when creating a new
...
+
+----
+"onFailure": [
+  {
+    "type": "StartBuild",
+    "build": {
+      "kind": "BuildConfig",
+      "name": "downstream-buildconfig-3", <1>
+      "namespace": "downstream-bc-namespace" <2>
+      },
```
Comment (php-coder) [inline dev_guide/builds.adoc]: I would rather remove extra commas.

**Comment by PI-Victor** [comment]
@php-coder thanks for the feedback, good suggestions, will update!

File: dev_guide/builds.adoc (hunk 3 above)
Comment (rhcarvalho) [inline dev_guide/builds.adoc]: extra commas are invalid JSON.
Will need to update to match the final implementation.

File: dev_guide/builds.adoc (hunk 2 above)
Comment (rhcarvalho) [inline dev_guide/builds.adoc]: Yes, it is optional. We should document that it defaults to the namespace of the upstream build.

**Comment by PI-Victor** [comment]
closing, this will be referenced in another PR.

**Pull request closed without merging**